baxter_collaboration_lib/test/test_ros_thread_image.cpp

```diff
@@ -16,14 +16,33 @@ class ROSThreadImageTester
         image_pub = it.advertise("/"+name+"/image", 1);
     }
 
-    void sendTestImage()
+    void sendTestImage(std::string _encoding = "bgr8")
     {
-        // Let's create a 200x200 black image with a red circle in the center
-        cv::Mat img(200, 200, CV_8UC3, cv::Scalar(0,0,0));
-        cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,0,0), -1);
+        // Let's create a 200x200 black image with a red circle
+        // (white circle for mono / 1-channel images) in the center
+        // Checks encoding and correctly creates an image with
+        // the requested encoding and publishes it
+        sensor_msgs::ImagePtr msg;
 
-        // Publish the image
-        sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", img).toImageMsg();
+        cv::Mat img;
+
+        if      (_encoding == "mono8")
+        {
+            img = cv::Mat(200, 200, CV_8UC1, cv::Scalar(0));
+            cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,255,255), -1);
```

Comment:
you should not assign a 3 channel color (ie `CV_RGB(255,255,255)`) to a mono image. Please fix